### PR TITLE
Only request client certificate when using certificate authentication

### LINKED ISSUE #3649: Web UI prompts for a certificate to authenticate
If i click cancel then i can enter user/pass in the login screen, but how to stop it prompting for 'certificate to authenticate' choice in first place?

image

### PR DESCRIPTION
**Pull request by dain**
Fixes #3649

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -46,7 +46,7 @@
         <dep.accumulo.version>1.7.4</dep.accumulo.version>
         <dep.accumulo-hadoop.version>2.7.7-1</dep.accumulo-hadoop.version>
         <dep.antlr.version>4.7.1</dep.antlr.version>
-        <dep.airlift.version>0.198</dep.airlift.version>
+        <dep.airlift.version>0.200-SNAPSHOT</dep.airlift.version>
```
Comment (electrum) [inline pom.xml:49]: Update before merging

File: presto-main/src/main/java/io/prestosql/server/security/ServerSecurityModule.java [hunk 2]
```diff
@@ -61,7 +63,10 @@ protected void setup(Binder binder)
 
         authenticatorBinder(binder); // create empty map binder
 
-        installAuthenticator("certificate", CertificateAuthenticator.class, CertificateConfig.class);
+        install(authenticatorModule("certificate", CertificateAuthenticator.class, certificateBinder -> {
+            newOptionalBinder(binder, ClientCertificate.class).setBinding().toInstance(ClientCertificate.REQUESTED);
```
Comment (electrum) [inline presto-main/src/main/java/io/prestosql/server/security/ServerSecurityModule.java:67]: Wrong binder

File: presto-main/src/main/java/io/prestosql/server/security/ServerSecurityModule.java [hunk 3]
```diff
@@ -61,7 +63,10 @@ protected void setup(Binder binder)
...
+            configBinder(certificateBinder).bindConfig((Class<?>) CertificateConfig.class);
```
Comment (electrum) [inline presto-main/src/main/java/io/prestosql/server/security/ServerSecurityModule.java:68]: Why do we need this cast?

**Comment by tooptoop4** [comment]
does it fix https://github.com/prestosql/presto/issues/3649 ?

**Comment by dain** [comment]
@tooptoop4 yep this fixes #3649

**Pull request merged**